lib/Sema/CSSimplify.cpp

```diff
@@ -9493,14 +9495,31 @@ performMemberLookup(ConstraintKind constraintKind, DeclNameRef memberName,
   if (isSingleUnlabeledPackExpansionTuple(instanceTy)) {
     auto elementTy = instanceTy->castTo<TupleType>()->getElementType(0);
     if (elementTy->is<TypeVariableType>()) {
-      MemberLookupResult result;
       result.OverallResult = MemberLookupResult::Unsolved;
       return result;
     }
   }
 
+  // Delay solving member constraint for unapplied methods
+  // where the base type has a conditional Sendable conformance
+  if (ctx.LangOpts.hasFeature(Feature::InferSendableMethods)) {
+    if (functionRefKind == FunctionRefKind::Unapplied) {
+      auto sendableProtocol =
+          DC->getParentModule()->getASTContext().getProtocol(
+              KnownProtocolKind::Sendable);
+      auto baseSendable = swift::TypeChecker::conformsToProtocol(
+          instanceTy, sendableProtocol, DC->getParentModule());
+
+      if (!baseSendable.isInvalid() &&
+          !baseSendable.getConditionalRequirements().empty() &&
+          instanceTy->hasTypeVariable()) {
```

Comment:
Nit: We could narrow this check slightly by checking any of the conditional requirements have a type variable on either side, instead of checking that the base type has a type variable. I don't think it'll make a difference in too many cases, though, because most conditional `Sendable` conformances like `extension Array: Sendable where Element: Sendable` or similar.